modules/data-entry/src/main/frontend/src/vocabQuery/VocabularyQuery.jsx

```diff
@@ -0,0 +1,389 @@
+//
+//  Licensed to the Apache Software Foundation (ASF) under one
+//  or more contributor license agreements.  See the NOTICE file
+//  distributed with this work for additional information
+//  regarding copyright ownership.  The ASF licenses this file
+//  to you under the Apache License, Version 2.0 (the
+//  "License"); you may not use this file except in compliance
+//  with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the License is distributed on an
+//  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+//  KIND, either express or implied.  See the License for the
+//  specific language governing permissions and limitations
+//  under the License.
+//
+import classNames from "classnames";
+import React, { useRef, useState } from "react";
+import PropTypes from "prop-types";
+
+import { withStyles, ClickAwayListener, Grow, IconButton, Input, InputAdornment, InputLabel, FormControl, Typography } from "@material-ui/core"
+import { LinearProgress, MenuItem, MenuList, Paper, Popper } from "@material-ui/core";
+
+import Search from "@material-ui/icons/Search";
+import Info from "@material-ui/icons/Info";
+
+import VocabularyBrowser from "./VocabularyBrowser.jsx";
+import { REST_URL, MakeRequest } from "./util.jsx";
+import QueryStyle from "./queryStyle.jsx";
+
+const NO_RESULTS_TEXT = "No results";
+const MAX_RESULTS = 10;
+
+// Component that renders a search bar for vocabulary terms.
+//
+// Required arguments:
+//  clearOnClick: Whether selecting an option will clear the search bar (default: true)
+//  onClick: Callback when the user clicks on this element
+//  focusAfterSelecting: focus after selecting (default: true)
+//  questionDefinition: Object describing the Vocabulary Question for which this suggested input is displayed
+//
+// Optional arguments:
+//  disabled: Boolean representing whether or not this element is disabled
+//  onInputFocus: Callback when the input is focused on
+//  label: Default text to display in search bar when nothing has been entered (default: 'Search')
+//  overrideText: When not undefined, this will overwrite the contents of the search bar
```

Comment:
```suggestion
//  overrideText: When not undefined, this will overwrite the contents of the search bar whenever the component is disabled. Useful, for example, to display a message like "Please select at most X options"
```

This would be more accurate. However, this prop is never set on the component, so it could be removed completely. We have a different mechanism for displaying the "select this many answers" message now.